simulator/infra/ports/t/unit_test.cpp

```diff
@@ -132,7 +200,42 @@ int main( int argc, char** argv)
 
 
 namespace ports {
-    namespace testing {
+    namespace testing {  
+
+        bool Logic::check_readiness( uint64 cycle, int check_code, bool is_ready)
+        {
+            assert( check_code >= 0 && check_code <= 3);
+
+            if ( cycle > EXPECTED_MAX_CYCLE)
+                return false;
+
+            if ( check_code == IS_STOP_READY)
+                return (cycle == EXPECTED_MAX_CYCLE) == is_ready;
+            else
+            {
+                bool is_ok = readiness_logic[cycle][check_code] == is_ready;
+                if ( is_ok && is_ready)
+                    readiness_logic[cycle][check_code] = false;
+
+                return is_ok;    
+            }
+        }
+
+        bool Logic::check_data( uint64 cycle, int check_code, int data)
+        {
+            assert( check_code >= 0 && check_code <= 2);
+
+            if ( cycle > EXPECTED_MAX_CYCLE || data < 0)
+                return false;
+
+            if ( check_code == DATA_INIT)
```

Comment:
For example, it takes some time to understand this conditions. Comments are needed here,

Reply:
I will add them.